src/main/java/org/jenkinsci/plugins/deployment/Condition.java

```diff
@@ -16,8 +16,12 @@
      * of the given {@link Job}. If so, return the build number of the job that matches,
      * otherwise return negative number to indicate no match.
      */
+    @Deprecated
     public abstract RangeSet calcMatchingBuildNumberOf(Job upstream, DeploymentFacet<?> facet);
 
+    public abstract RangeSet calcMatchingBuildNumberOf(Job upstream, DeploymentFacet<?> facet, Job workflowJob);
```

Comment:
Backward incompatibility (this is going to break any implementation). You should implement something [like this](https://github.com/jenkinsci/analysis-core-plugin/blob/9f8f492d2931981f91a06a8eb0123fd7d3cd1a58/src/main/java/hudson/plugins/analysis/core/HealthAwarePublisher.java#L129-L145).